calico_containers/pycalico/ipam.py

```diff
@@ -782,6 +820,56 @@ def get_assignment_attributes(self, address):
             _, attributes = block.get_attributes_for_ip(address)
             return attributes
 
+    def get_cidr_assignment_attributes(self, cidr):
+        """
+        Return the attributes of a given CIDR. Depending on the size of the
+        CIDR in question, this could have a very large result.
+
+        :param cidr: CIDR to query.
+        :type cidr: IPNetwork
+        :return: The attributes of all the addresses in a given CIDR.
+        :rtype: list
+        """
+        assert isinstance(cidr, IPNetwork)
+
+        # List of IPNetworks corresponding to blocks in ETCD
+        block_cidrs = self._get_all_blocks(cidr)
+
+        # List of AllocationBlocks
+        blocks = []
+        for block_cidr in block_cidrs:
+            block = self._read_block(block_cidr)
+            blocks.append(block)
+
+        # List of IPPools
+        pools = self.get_ip_pools(cidr.version)
+
+        result = []
+        for block in blocks:
+            for pool_candidate in pools:
+                if block in pool_candidate:
+                    pool = str(pool_candidate)
+                    break
+
+            for num, index in enumerate(block.allocations):
```

Comment:
I suggest you move this code into block.py.  It works with internals of the `AllocationBlock` class and is nice and self contained to a single block.  That way we keep the interface to 'AllocationBlock` abstract from its implementation.